### ISSUE TITLE
Tilde not loaded as null in decode

### ISSUE THREAD

**Issue by StefanFabian** [jbeder/yaml-cpp#860]
```
first: ~
second: "A string without tag"
```
Using the templated converter decode method I noticed that first is not decoded as a Null node but a Scalar "~".
In #357 this seems to be fixed but apparently with `YAML::LoadFile` it doesn't work (anymore?).

Thank you for your work on this btw. Overall it works great and is pretty easy to use and extend.

**Comment by jbeder** [comment]
Could you give some example code that demonstrates this bug?

**Comment by dota17** [comment]
It looks like  ```YAML::LoadFile```  work well and generates the node whose nodetype is NodeType::Null for the ```~```. 
https://github.com/jbeder/yaml-cpp/blob/a98b8af448b2140dd4aebca2c16fe6c25a82be23/src/nodebuilder.cpp#L37

And it will be outputted as ```~```
https://github.com/jbeder/yaml-cpp/blob/a98b8af448b2140dd4aebca2c16fe6c25a82be23/src/emitter.cpp#L890

**Comment by StefanFabian** [comment]
Using libyaml-cpp-dev 0.5.2 on Ubuntu 18.04 the following minimal example reproduces the bug on my pc:
test.yaml
```yaml
first: ~
second: "A string without tag"

```

yaml_test.cpp
```cpp
#include <yaml-cpp/yaml.h>
#include <iostream>

void dumpNode(const YAML::Node &node)
{
  switch (node.Type())
  {
    case YAML::NodeType::Null:
      std::cout << "Null" << std::endl;
      return;
    case YAML::NodeType::Map:
      std::cout << "Map:" << std::endl;
      for (auto it = node.begin(); it != node.end(); ++it)
      {
        std::cout << it->first.as<std::string>() << ":";
        dumpNode(it->second);
      }
      return;
    case YAML::NodeType::Scalar:
      std::cout << "Scalar: " << node.Scalar() << std::endl;
      return;
  }
}

class SomeClass {};

template <>
struct YAML::convert<SomeClass>
{

  static bool decode(const YAML::Node &node, SomeClass &out)
  {
    switch (node.Type())
    {
      case YAML::NodeType::Null:
        std::cout << "Null" << std::endl;
        return true;
      case YAML::NodeType::Map:
        std::cout << "Map:" << std::endl;
        for (auto it = node.begin(); it != node.end(); ++it)
        {
          std::cout << it->first.as<std::string>() << ":";
          it->second.as<SomeClass>();
        }
        return true;
      case YAML::NodeType::Scalar:
        std::cout << "Scalar: " << node.Scalar() << std::endl;
        return true;
    }
    return false;
  }
};

int main(int argc, char **argv)
{
  YAML::Node node = YAML::LoadFile("test.yaml");
  dumpNode(node);
  std::cout << std::endl << std::endl << "Using decoder:" <<std::endl;
  node.as<SomeClass>();
}

```
Compiling with g++ 7.5.0 and running
```
g++ yaml_test.cpp -lyaml-cpp && ./a.out
```
produces the following output:
```
Map:
first:Scalar: ~
second:Scalar: A string without tag


Using decoder:
Map:
first:Scalar: ~
second:Scalar: A string without tag
```

**Comment by dota17** [comment]
@StefanFabian I tested it in master branch (0.5.2 version building failed 0.0) and got the 
```
Map:
first:Null
second:Scalar: A string without tag

Using decoder:
Map:
first:Null
second:Scalar: A string without tag
```
It looks like this issue has been fixed as you said. But the ```YAML::LoadFile``` works well. So I am a little confused with 
> In #357 this seems to be fixed but apparently with YAML::LoadFile it doesn't work (anymore?).

**Comment by StefanFabian** [comment]
Are you saying it works for you with 0.5.2 or it works on the current master?

**Comment by jbeder** [comment]
I didn't look for when this was fixed, but it certainly has the right behavior for 0.6.0 and on.

The commit bfb5703ce92b18e1f2d974a7c567a003184b64b0 added a test [here](https://github.com/jbeder/yaml-cpp/blob/a98b8af448b2140dd4aebca2c16fe6c25a82be23/test/integration/load_node_test.cpp#L250).

If you're still on 0.5.2, I recommend updating to latest (right now 0.6.3); 0.5.2 was released in 2015.

**Comment by StefanFabian** [comment]
Unfortunately, I'm stuck with what's available in the default repositories on Ubuntu 18.04.
